### DISCUSSION TITLE
IINA 1.3.2

### DISCUSSION THREAD

**Discussion by lhc70000** [iina/iina#4479] [Announcements]
# New
- Added a new Log Viewer. You can view logs more easily by pressing ⇧+⌘+L [#4109](https://github.com/iina/iina/pull/4109)
- Added HDR tone mapping settings #[4358](https://github.com/iina/iina/issues/4358)
- Added support for VP9 hardware acceleration #[4149](https://github.com/iina/iina/issues/4149)
- Added support for Apple ProRes hardware acceleration #[3874](https://github.com/iina/iina/issues/3874)
- Added  `ogv` and `ogm` to the supported video file extensions list [#4274](https://github.com/iina/iina/pull/4274)
- Added A-B Loop indicators on progress bar #[548](https://github.com/iina/iina/issues/548) 
- Additional information is shown including FFmpeg version in the about window [#3977](https://github.com/iina/iina/issues/3977)
- Updated document icons to follow Apple's latest conventions [#3670](https://github.com/iina/iina/discussions/3670)
- Build date, git branch and commit hash are now included in the log file #[3970](https://github.com/iina/iina/issues/3970)

# Bug Fixes
- Fixed no audio while streaming YouTube videos [#4253](https://github.com/iina/iina/issues/4253)
- Fixed incorrect rendering of some HDR videos [#3772](https://github.com/iina/iina/issues/3772)
- Fixed Playback/Timebar freezes at the start while playing any audio file [#4051](https://github.com/iina/iina/issues/4051)
- Fixed an issue where timeline in Touch Bar sometimes freezes [#4058](https://github.com/iina/iina/issues/4058)
- Fixed Playlist/Chapters no longer shows automatically in Music mode [#4088](https://github.com/iina/iina/issues/4088)
- Fixed black window when switching between video and music mode [#4055](https://github.com/iina/iina/issues/4055) and PiP [#4268](https://github.com/iina/iina/issues/4268)
- Fixed lag when scrolling to seek [#4153](https://github.com/iina/iina/issues/4153)
- Fixed serious lag when playing audio [#4049](https://github.com/iina/iina/issues/4049)
- Fixed a crash during quit when in full screen [#4020](https://github.com/iina/iina/issues/4020)
- Fixed IINA crashed when quitting [#4045](https://github.com/iina/iina/issues/4045)
- Fixed crash in `VideoView.requestEdrMode` while quitting [#4328](https://github.com/iina/iina/issues/4328)
- Fixed crash in `mpv_render_context_report_swap` while quitting [#4315](https://github.com/iina/iina/issues/4315)
- Fixed "Problem Report for IINA" pop-up everytime [#4182](https://github.com/iina/iina/issues/4182)
- Fixed crash in `mpv_set_property` during termination [#3596](https://github.com/iina/iina/issues/3596)
- Fixed assert from mpv, !queue->in_process, during quit [#3824](https://github.com/iina/iina/issues/3824)
- Fixed IINA stopped responding after quit clicked [#4310](https://github.com/iina/iina/issues/4310)
- Fixed IINA sometimes takes a few seconds to quit [#4227](https://github.com/iina/iina/issues/4227)
- Fixed a crash when passed '-' in the parameter list [#4022](https://github.com/iina/iina/issues/4022)
- Fixed a crash when capturing a screenshot with a long video title [#3334](https://github.com/iina/iina/issues/3334)
- Fixed a crash related to `MPNowPlayingInfoCenter` [#4251](https://github.com/iina/iina/issues/4251)
- Fixed a crash when setting the preferred language for subtitles [#4074](https://github.com/iina/iina/pull/4074)
- Fixed inspector displays "Error" for filename [#4289](https://github.com/iina/iina/issues/4289)
- Fixed sometimes wrong title is shown in the window menu [#4229](https://github.com/iina/iina/issues/4229)
- Fixed some issues when resizing the custom crop [#4299](https://github.com/iina/iina/issues/4299)
- Fixed an issue where blur filter cannot be applied with max matrix size [#4259](https://github.com/iina/iina/issues/4259)
- Fixed an issue where full screen mode only applies to the first opened video [#4202](https://github.com/iina/iina/issues/4202)
- Fixed filters cannot be applied to multiple windows [#4273](https://github.com/iina/iina/issues/4273)
- Fixed deleting files in the playlist via context menu may cause playlist item mismatch [#4061](https://github.com/iina/iina/issues/4061)
- Fixed subtitles panel shows `...` for subtitle number [#4341](https://github.com/iina/iina/issues/4341)
- Fixed sidebar active tab button colors for do not match [#4332](https://github.com/iina/iina/issues/4332)
- Fixed numbers in menu items do not change back to defaults [#4221](https://github.com/iina/iina/issues/4221)
- Fixed cursor remains visible after resuming playback [#4183](https://github.com/iina/iina/issues/4183)
- Fixed preferences' first item is too close to the top [#4100](https://github.com/iina/iina/issues/4100)
- Fixed unable to hide Quick Panel with On Screen Controller [#4071](https://github.com/iina/iina/issues/4071)
- Fixed "mpv Default" keyboard shortcuts display discrepancies in Preferences/Playback [#4050](https://github.com/iina/iina/issues/4050)
- Fixed layout errors in Settings > UI tab [#4033](https://github.com/iina/iina/issues/4033)
- Fixed `Show Quick Settings Panel` menu item to toggle to `Hide Quick Settings Panel` [#3853](https://github.com/iina/iina/issues/3853)
- Fixed preference panes keep shifting vertically [#3717](https://github.com/iina/iina/issues/3717)
- Fixed crash due to too many update constraints [#3505](https://github.com/iina/iina/issues/3505)
- Fixed crash in `objc_release` called from `PlayerCore.fileStarted` [#3822](https://github.com/iina/iina/issues/3822)
- Fixed saved filter key shortcuts: lowercase letters have shift modifier added [#3998](https://github.com/iina/iina/issues/3998)

# Improvements
- Resolve symlinks in recently opened files to prevent duplicates [#3788](https://github.com/iina/iina/issues/3788)
- Controls in the video panel now don't overlap [#3790](https://github.com/iina/iina/issues/3790)
- Replace `current-window-scale` with `window-scale` in the default mpv input conf [#4053](https://github.com/iina/iina/issues/4053)
- Various UI improvements for preference window to improve consistency and reliability [#4107](https://github.com/iina/iina/pull/4107)
- OSD progress bar now grows with the text length [#3171](https://github.com/iina/iina/issues/3171)
- Using delete key to remove entries in the history window now gives warning [#4255](https://github.com/iina/iina/issues/4255)
- Audio menu now reflects mute state [#4201](https://github.com/iina/iina/issues/4201)
- The Toolbar Settings Sheet Window can now respond to Enter and ESC keys.
- Image based subtitles can be scaled using the slider in the quick settings panel [#4225](https://github.com/iina/iina/issues/4225)
- Improve the mouse drag detection in the main player window [#4231](https://github.com/iina/iina/issues/4231)
- Overhaul to the playback history window [#4257](https://github.com/iina/iina/pull/4257)
- `Reveal in Finder` renamed to `Show in Finder` [#3976](https://github.com/iina/iina/issues/3976)
- Disclosure triangles obey reduced motion macOS setting #[4370](https://github.com/iina/iina/issues/4370)
- Entering and exiting interactive view obeys reduced motion macOS setting [#4301](https://github.com/iina/iina/issues/4301)

# Updates
- Updated [mpv](https://mpv.io/) to 0.35.1, adding patches for the following mpv issues:
    - EDL error when trying to play MP4 DASH videos from YouTube with yt-dlp #[11392](https://github.com/mpv-player/mpv/issues/11392)
    - While paused mpv prevents sleep on Mac, consumes CPU #[11617](https://github.com/mpv-player/mpv/issues/11617)
    - Streaming from URL ending in / causes memory corruption #[11626](https://github.com/mpv-player/mpv/issues/11626)
- Updated [FFmpeg](https://ffmpeg.org/) to 6.0
- Updated [yt-dlp](https://github.com/yt-dlp/yt-dlp) to 2023.06.22
- Updated [Sparkle](https://sparkle-project.org/) to 2.4.0, fix [#4046](https://github.com/iina/iina/issues/4046)

<hr /><em>This discussion was created from the release <a href='https://github.com/iina/iina/releases/tag/v1.3.2'>IINA 1.3.2</a>.</em>

**Comment by adriel** [comment]
Wow this is a huge update, so many improvements. Thank you to all involved!!

**Comment by AlejandroVolkova** [comment]
Thanks for the update and bug fixes guys, I appreciate your hard work, specially for fixing the issue I reported #4153.
However for some reason **IINA 1.3.2** gets frozen when playing _.webm_ files, and some _.mkv_ files.  which is weird because It doesn’t happens with previous versions of IINA.

I created a completely new and clean admin account to test if it was some 3rd party app generating conflict but it wasn’t the case. of course I tested the same files before mentioned.

**Reply by lhc70000** [reply]
I just tried the a webm file and IINA worked well. Could you please share a problematic file with us to reproduce the issue?

**Reply by AlejandroVolkova** [reply]
Sure , [here](https://www.dropbox.com/sh/x5go7s718tbopen/AADVhpJh-_UUaXYfU2rDoeoFa?dl=0) is a .webm file that completely freezes IINA , it happens with most of my .webm files and some mkv files. (none of them freeze previous versions of IINA)

<img width="564" alt="IINA-1 3 2" src="https://github.com/iina/iina/assets/120342552/679d3678-25a1-4893-9000-e10a8faee6bc">


My Mac specs are : 

iMac Retina 4K, 21.5-Inch, 2019
**Processor :** 3 GHz 6-Core Intel Core i5
**Graphics :**  Radeon Pro 560X 4 GB
**Memory :** 32 GB 2667 MHz DDR4
**macOS :**  Ventura 13.4.1

**Reply by ejkarne** [reply]
Chiming in to say this is likely caused by [VideoToolbox VP9 hwaccel freezes ffmpeg](https://trac.ffmpeg.org/ticket/9599). A workaround is adding the mpv option `hwdec-codecs` and setting it to `h264,vc1,hevc,vp8,av1` (removing VP9) in the Advanced menu in the settings.

**Reply by AlejandroVolkova** [reply]
Thank you so much, I think you are absolutely right because this workaround fixed my issues , now I can play the videos that I could not before.

**Reply by AlejandroVolkova** [reply]
After doing some tests with the information provided by @GreyAsteroid  I can confirm that the videos that were freezing **IINA 1.3.2** are videos under the _**VP9 Codec**_. Mostly sourced from _YouTube_. 

Here is a _.mkv_ video under **AOMedia** codec which has no issues at all : 

<img width="552" alt="MKV (AOMedia) No Issues : " src="https://github.com/iina/iina/assets/120342552/9a0ab05a-5224-41e3-9206-8b3f69543fee">

on the other side here's a _.mkv_ file under **VP9 Codec** which completely freezes IINA :

<img width="552" alt="Webm (VP90) Issues" src="https://github.com/iina/iina/assets/120342552/9afb2c78-d066-42d2-add6-a52d4300b362">

So, we can determine that the problem is not related to the file extension, but rather to the VP9 codec.

**Reply by newtrashcan** [reply]
I have this problem too. Cannot play webm files in 1.3.2. They worked in 1.3.1.
iMac 19,1 macOS 12.6.5.
As GreyAsteroid said above, the problem in hardware acceleration for vp9 codec.
I suggest to make video decoding usng sjftware decocer - default option and make some checkbox for each decoder to manually enable hardware acceleration.
Thank you.

**Reply by uiryuu** [reply]
Anyone who has issue playing VP9 codec files, please test this out. We've disabled hardware decoding on Intel Macs for VP9 codec:
https://drive.google.com/file/d/1c4V57M4rwJzlzjEHM4saI6yzMTarvG_3/view?usp=drive_link

**Reply by AlejandroVolkova** [reply]
@uiryuu I can confirm this build works perfectly right out of the box on Intel-based Macs that has issues playing VP9 codec files (of course without using any workaround).

**Reply by jzrskc** [reply]
I have the samo problem with the webm format on the version: IINA 1.3.2

**Reply by low-batt** [reply]
@jzrskc Please upgrade to the just released [v1.3.3](https://github.com/iina/iina/releases/tag/v1.3.3). This release contains a workaround for the FFmpeg freeze that disables VP9 hardware decoding on Intel Macs, issue #4486.

**Comment by lhc70000** [comment]
I have updated the dmg file (build 135) to fix the missing document icons. Please re-download it from the GitHub release page or https://iina.io.

Auto-update to 1.3.2 is temporarily disabled until our mirror site is synced.

**Comment by konrad-a** [comment]
It is not the best idea to release different binaries under the same version number.
Now we have 2 binary versions of 1.3.2 in the Internet.
For example, the brew is broken:
```==> Upgrading 1 outdated package:
iina 1.3.1 -> 1.3.2
==> Upgrading iina
==> Downloading https://github.com/iina/iina/releases/download/v1.3.2/IINA.v1.3.2.dmg
Already downloaded: /Users/.../Library/Caches/Homebrew/downloads/2b39d99b58afc65da13d09c274198206bcd2652ed7e0ed5bf8774ed9333238de--IINA.v1.3.2.dmg
==> Purging files for version 1.3.2 of Cask iina
Error: iina: SHA256 mismatch
Expected: ca278070c737aeaa1446c01c73c8806ded69a21940537afbe85b2051e88d4cff
  Actual: ac5e72bf6407595514b327ffbbbf235919f80badbd18126007f8bc62768a1c19
    File: /Users/.../Library/Caches/Homebrew/downloads/2b39d99b58afc65da13d09c274198206bcd2652ed7e0ed5bf8774ed9333238de--IINA.v1.3.2.dmg
To retry an incomplete download, remove the file above.
```


It would be better to release it as 1.3.2.1 patch.

**Comment by LangLangBart** [comment]
PS: Thanks to @low-batt for the `--music-mode` flag in `1.3.2` d1fec00f531edb536e6407d1de198015c0820468 

![gifsicle](https://github.com/iina/iina/assets/92653266/9e894666-768b-401c-b2a5-65787f2686d9)

**Comment by low-batt** [comment]
I have created issue #4486 for the VP9 related freeze.

**Comment by raisin8010** [comment]
I think 1.3.2 need to be republished, for there are many strings lost localization. Like in #4481, and some other text in preferences.

**Comment by shinokiri** [comment]
1.3.2 also fixed 10-bit color output. Not sure if it's from upstream as it's not listed in the changelog.
And not clear when the problem was introduced, or if it's the first time IINA has supported 10-bit color.
I found an earlier issue about this #3528 

Decoding performance has indeed been enhanced. Now I can easily hardware decode h264 yuv444p10 format. This should come from upstream, since the P410 biplanar format used by VideoToolbox HW decoder was only [added after FFmpeg 5.0](https://github.com/FFmpeg/FFmpeg/commit/d2229eca5138a9974212f9ef4e7a2eb3d4e04025).

**Comment by low-batt** [comment]
@shinlw, There were two IINA changes related to HDR.  From the release notes:
- Added HDR tone mapping settings #4358
- Fixed incorrect rendering of some HDR videos #3772

Likely the second one improved color handling. That change is in PR #4174. The new tone mapping settings are disabled by default. They are found under `Audio/Video` in IINA's settings.

I'm still trying to understand what the situation is with HDR running on a Mac with the M2 chip. Many users reported the dark screen and flashing that happens with IINA 1.3.1 was not occuring with a recent nightly build. But a couple of M2 owners reported it was not fixed.

And yes the mpv and FFmpeg upgrades have picked up improvements in hardware decoding, although as you can see above there is trouble with VP9 hardware decoding under Intel Macs.

**Reply by shinokiri** [reply]
Actually I was just testing 10bit in SDR, with some simple pics and video like this. So probably the floating pixel format is to be credited.

https://github.com/iina/iina/assets/45729567/fcc401f4-8bb1-43fe-a1e8-50624b4f8338

BTW, this video can also confirm a screenshot bug, which will appear when the transfer characteristics of a video is tagged as sRGB.
